platform-sdk/swirlds-logging-log4j-appender/src/test/java/com/swirlds/logging/log4j/appender/SwirldsLogAppenderTest.java

```diff
@@ -0,0 +1,191 @@
+
+import com.swirlds.config.api.Configuration;
+import com.swirlds.config.extensions.test.fixtures.TestConfigBuilder;
+import com.swirlds.logging.api.Level;
+import com.swirlds.logging.api.internal.LoggingSystem;
+import com.swirlds.logging.legacy.LogMarker;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Stream;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.apache.logging.log4j.Marker;
+import org.apache.logging.log4j.MarkerManager;
+import org.apache.logging.log4j.ThreadContext;
+import org.apache.logging.log4j.core.LoggerContext;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.Arguments;
+import org.junit.jupiter.params.provider.MethodSource;
+
+class SwirldsLogAppenderTest {
+    private static final String LOGGER_NAME = "testLogger";
+
+    public LoggingSystem setup(final com.swirlds.config.api.Configuration configuration) {
+        final LoggingSystem loggingSystem = new LoggingSystem(configuration);
+
+        loggingSystem.installHandlers();
+        loggingSystem.installProviders();
+
+        return loggingSystem;
+    }
+
+    static Stream<Arguments> levelsAndSizes() {
+        return Stream.of(
+                Arguments.of(Level.TRACE, 5),
+                Arguments.of(Level.DEBUG, 4),
+                Arguments.of(Level.INFO, 3),
+                Arguments.of(Level.WARN, 2),
+                Arguments.of(Level.ERROR, 1),
+                Arguments.of(Level.OFF, 0));
+    }
+
+    @ParameterizedTest
+    @MethodSource("levelsAndSizes")
+    void loggingLevels(final Level level, final int expectedSize) throws Exception {
+        // given
+        final Path filePath = Path.of("%s_test.log".formatted(level));
+        try {
+            final LoggingSystem loggingSystem = setup(createConfig(level, filePath));
+            final Logger testLogger = LogManager.getLogger(LOGGER_NAME);
+
+            // when
+            testLogger.error("This is an error message");
+            testLogger.warn("This is a warn message");
+            testLogger.info("This is an info message");
+            testLogger.debug("This is a debug message");
+            testLogger.trace("This is a trace message");
+
+            // then
+            loggingSystem.stopAndFinalize();
+            final List<String> logLines = Files.lines(filePath).toList();
+
+            assertThat(logLines).hasSize(expectedSize);
+        } finally {
+            Files.delete(filePath);
+        }
+    }
+
+    @Test
+    void loggingSimpleMarker() throws Exception {
+        // given
+        final Path filePath = Path.of("marker_test.log");
+        try {
+            final LoggingSystem loggingSystem = setup(createConfig(Level.DEBUG, filePath));
+            final Logger testLogger = LogManager.getLogger(LOGGER_NAME);
```

Comment:
it would be way easier to read if you call it log4JLogger instead of test logger.